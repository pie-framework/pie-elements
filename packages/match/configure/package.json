{
  "name": "@pie-element/match-configure",
  "private": true,
  "version": "3.1.3",
  "description": "",
  "main": "lib/index.js",
  "author": "",
  "dependencies": {
    "@material-ui/core": "^3.9.2",
    "@material-ui/icons": "^3.0.1",
    "@pie-framework/pie-configure-events": "^1.2.0",
    "@pie-lib/config-ui": "^10.7.34",
    "@pie-lib/drag": "^1.1.21",
    "@pie-lib/editable-html": "^7.10.24",
    "@pie-lib/scoring-config": "^3.5.40",
    "classnames": "^2.2.5",
    "debug": "^3.1.0",
    "prop-types": "^15.6.2",
    "react": "^16.8.1",
    "react-dnd": "^9.3.2",
    "react-dom": "^16.8.1",
    "react-swipeable-views": "^0.12.17"
  },
  "license": "ISC"
}
