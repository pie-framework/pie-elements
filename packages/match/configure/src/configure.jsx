import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import {
  FeedbackConfig,
  settings,
  layout,
  InputContainer
} from '@pie-lib/config-ui';
import EditableHtml from '@pie-lib/editable-html';
import PropTypes from 'prop-types';
import debug from 'debug';
import Typography from '@material-ui/core/Typography';
import GeneralConfigBlock from './general-config-block';
import AnswerConfigBlock from './answer-config-block';

const log = debug('@pie-element:match:configure');
const { Panel, toggle, radio } = settings;

const styles = theme => ({
  title: {
    fontSize: '1.1rem',
    display: 'block',
    marginTop: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit
  },
  content: {
    marginTop: theme.spacing.unit * 2
  },
  promptHolder: {
    width: '100%',
    paddingTop: theme.spacing.unit * 2
  },
  prompt: {
    paddingTop: theme.spacing.unit * 2,
    width: '100%'
  },
});

class Configure extends React.Component {
  static propTypes = {
    onModelChanged: PropTypes.func,
    onConfigurationChanged: PropTypes.func,
    classes: PropTypes.object,
    model: PropTypes.object.isRequired,
    configuration: PropTypes.object.isRequired,
    imageSupport: PropTypes.shape({
      add: PropTypes.func.isRequired,
      delete: PropTypes.func.isRequired
    })
  };

  static defaultProps = {
    classes: {}
  };

  constructor(props) {
    super(props);

    this.rowIdCounter = props.model.rows[props.model.rows.length - 1].id + 1;

    this.state = {
      activeTab: 0
    };
  }

  onTabChange = (event, value) => {
    this.setState({ activeTab: value });
  };

  onChangeTabIndex = index => {
    this.setState({ activeTab: index });
  };

  onChange = model => {
    this.props.onModelChanged(model);
  };

  onFeedbackChange = feedback => {
    const { model, onModelChanged } = this.props;
    model.feedback = feedback;
    onModelChanged(model);
  };

  onDeleteRow = (rowIndex) => {
    const { model } = this.props;

    const newModel = { ...model };

    newModel.rows.splice(rowIndex, 1);

    this.onChange(newModel);
  };

  onAddRow = () => {
    const { model } = this.props;
    const newModel = { ...model };

    newModel.rows = newModel.rows.concat({
      id: this.rowIdCounter + 1,
      title: `Question Text ${newModel.rows.length + 1}`,
      values: new Array(model.layout - 1).fill(false)
    });

    this.rowIdCounter += 1;

    this.onChange(newModel);
  };

  onLayoutChange = (newLayout) => {
    const { model } = this.props;
    const oldLayout = model.layout;
    const newModel = { ...model };

    if (newLayout > oldLayout) {
      for (let i = 0; i < (newLayout - oldLayout); i++) {
        newModel.headers.push(`Column ${newModel.headers.length + 1}`);
      }

      newModel.rows.forEach(row => {
        for (let i = 0; i < (newLayout - oldLayout); i++) {
          row.values.push(false);
        }
      });
    } else if (newLayout < oldLayout) {
      newModel.headers.splice(newLayout);

      newModel.rows.forEach(row => {
        row.values.splice(newLayout - 1);
      });

    }

    newModel.layout = newLayout;

    this.onChange(newModel);
  };

  onResponseTypeChange = newChoiceMode => {
    const { model } = this.props;
    const newModel = { ...model };

    // if we're switching to radio and we have more than one true, reset
    if (newChoiceMode === 'radio') {
      newModel.rows.forEach(row => {
        const trueCount = row.values.reduce((total, current) => current === true ? total + 1 : total);

        if (trueCount > 1) {
          row.values = new Array(model.layout - 1).fill(false)
        }
      })
    }

    newModel.choiceMode = newChoiceMode;

    this.onChange(newModel);
  };

  onPartialScoringChange = partialScoring => {
    this.props.model.partialScoring = partialScoring;
    this.props.onModelChanged(this.props.model);
  };

  onPromptChanged = prompt => {
    this.props.onModelChanged({
      ...this.props.model,
      prompt
    });
  };

  onTeacherInstructionsChanged = teacherInstructions => {
    this.props.onModelChanged({
      ...this.props.model,
      teacherInstructions
    });
  };

  onRationaleChanged = rationale => {
    this.props.onModelChanged({
      ...this.props.model,
      rationale
    });
  };

  render() {
    const { classes, model, imageSupport, onModelChanged, configuration, onConfigurationChanged } = this.props;
    const {
      enableImages,
      partialScoring,
      teacherInstructions = {},
      studentInstructions,
      rationale = {},
      lockChoiceOrder,
      scoringType,
      prompt,
    } = configuration;

    log('[render] model', model);


    return (
      <layout.ConfigLayout
        settings={
          <Panel
            model={model}
            configuration={configuration}
            onChangeModel={model => onModelChanged(model)}
            onChangeConfiguration={config => onConfigurationChanged(config)}
            groups={{
              'Settings': {
                enableImages: enableImages.settings &&
                toggle(enableImages.label),
                partialScoring: partialScoring.settings &&
                toggle(partialScoring.label),
                lockChoiceOrder: lockChoiceOrder.settings &&
                toggle(lockChoiceOrder.label)
              },
              'Properties': {
                'teacherInstructions.enabled': teacherInstructions.settings &&
                toggle(teacherInstructions.label, true),
                'studentInstructions.enabled': studentInstructions.settings &&
                toggle(studentInstructions.label, true),
                'rationale.enabled': rationale.settings &&
                toggle(rationale.label, true),
                scoringType: scoringType.settings &&
                radio(scoringType.label, ['auto', 'rubric']),
              },
            }}
          />
        }
      >
        <div className={classes.content}>
          <Typography component="div" type="body1">
            <span>
              In Choice Matrix, students associate choices in the first column with options in the adjacent rows.
              This interaction allows for either one or more correct answers. Setting more than one answer as correct allows for partial credit <i>(see the Scoring tab)</i>.
            </span>
          </Typography>

          {teacherInstructions.enabled && (
            <InputContainer label={teacherInstructions.label} className={classes.promptHolder}>
              <EditableHtml
                className={classes.prompt}
                markup={model.teacherInstructions || ''}
                onChange={this.onTeacherInstructionsChanged}
                imageSupport={imageSupport}
                nonEmpty={false}
              />
            </InputContainer>
          )}

          {prompt.settings && (
            <InputContainer
              label={prompt.label}
              className={classes.promptHolder}
            >
              <EditableHtml
                className={classes.prompt}
                markup={model.prompt}
                onChange={this.onPromptChanged}
                imageSupport={imageSupport}
                nonEmpty={!prompt.settings}
                disableUnderline
              />
            </InputContainer>
          )}

          {rationale.enabled && (
            <InputContainer label={rationale.label}
                            className={classes.promptHolder}>
              <EditableHtml
                className={classes.prompt}
                markup={model.rationale || ''}
                onChange={this.onRationaleChanged}
                imageSupport={imageSupport}
              />
            </InputContainer>
          )}
          <GeneralConfigBlock
            model={model}
            configuration={configuration}
            onResponseTypeChange={this.onResponseTypeChange}
            onLayoutChange={this.onLayoutChange}
          />
          <AnswerConfigBlock
            model={model}
            configuration={configuration}
            imageSupport={imageSupport}
            onChange={this.onChange}
            onAddRow={this.onAddRow}
            onDeleteRow={this.onDeleteRow}
          />
          <FeedbackConfig
            feedback={model.feedback}
            onChange={this.onFeedbackChange}
          />
        </div>

      </layout.ConfigLayout>
    );
  }
}

export const Config = Configure;

export default withStyles(styles)(Configure);
