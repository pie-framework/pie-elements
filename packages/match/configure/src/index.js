import React from 'react';
import ReactDOM from 'react-dom';
import Configure from './configure';
import {
  ModelUpdatedEvent,
  DeleteImageEvent,
  InsertImageEvent
} from '@pie-framework/pie-configure-events';
import debug from 'debug';
import defaultValues from './defaults';

const log = debug('pie-elements:match:configure');

export default class MatchConfigure extends HTMLElement {
  static createDefaultModel = (model = {}) => ({
      ...defaultValues.model,
      ...model
    }
  );

  constructor() {
    super();
    this._model = MatchConfigure.createDefaultModel();
    this._configuration = defaultValues.configuration;
  }

  set model(m) {
    this._model = MatchConfigure.createDefaultModel(m);
    this._render();
  }

  set configuration(c) {
    this._configuration = c;
    this._render();
  }

  onModelChanged(model) {
    this._model = model;
    log('[onModelChanged]: ', this._model);
    this._render();
    this.dispatchEvent(new ModelUpdatedEvent(this._model, true));
  }

  onConfigurationChanged(config) {
    this._configuration = config;
    this._render();
  }

  /**
   *
   * @param {done, progress, file} handler
   */
  insertImage(handler) {
    this.dispatchEvent(new InsertImageEvent(handler));
  }

  onDeleteImage(src, done) {
    this.dispatchEvent(new DeleteImageEvent(src, done));
  }

  _render() {
    if (this._model) {
      const el = React.createElement(Configure, {
        onModelChanged: this.onModelChanged.bind(this),
        onConfigurationChanged: this.onConfigurationChanged.bind(this),
        model: this._model,
        configuration: this._configuration,
        imageSupport: {
          add: this.insertImage.bind(this),
          delete: this.onDeleteImage.bind(this)
        }
      });
      ReactDOM.render(el, this);
    }
  }
}
