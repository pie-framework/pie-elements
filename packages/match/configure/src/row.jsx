import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import { Checkbox } from '@pie-lib/config-ui';
import DragHandle from '@material-ui/icons/DragHandle';
import Radio from '@material-ui/core/Radio';
import Button from '@material-ui/core/Button';
import Delete from '@material-ui/icons/Delete';
import { DragSource, DropTarget } from 'react-dnd';
import debug from 'debug';
import EditableHtml from '@pie-lib/editable-html';

const log = debug('@pie-element:categorize:configure:choice');

export let canDrag = false;

export class Row extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    model: PropTypes.object.isRequired,
    row: PropTypes.object.isRequired,
    idx: PropTypes.number.isRequired,
    isDragging: PropTypes.bool.isRequired,
    onDeleteRow: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    connectDragSource: PropTypes.func.isRequired,
    connectDropTarget: PropTypes.func.isRequired,
    onMoveRow: PropTypes.func.isRequired,
    imageSupport: PropTypes.shape({
      add: PropTypes.func.isRequired,
      delete: PropTypes.func.isRequired
    })
  };

  static defaultProps = {};

  componentDidMount() {
    document.addEventListener('mouseup', this.onMouseUpOnHandle);
  }

  onRowTitleChange = rowIndex => value => {
    const { model, onChange } = this.props;
    const newModel = { ...model };

    newModel.rows[rowIndex].title = value;

    onChange(newModel);
  };

  onRowValueChange = (rowIndex, rowValueIndex) => event => {
    const { model, onChange } = this.props;
    const newModel = { ...model };

    if (model.responseType === 'radio') {
      for (let i = 0; i < newModel.rows[rowIndex].values.length; i++) {
        newModel.rows[rowIndex].values[i] = false;
      }
    }

    newModel.rows[rowIndex].values[rowValueIndex] = event.target.checked;

    onChange(newModel);
  };

  onDeleteRow = (idx) => () => {
    this.props.onDeleteRow(idx)
  };

  onMouseDownOnHandle = () => {
    canDrag = true;
  };

  onMouseUpOnHandle = () => {
    canDrag = false;
  };

  render() {
    const {
      classes,
      imageSupport,
      connectDragSource,
      connectDropTarget,
      isDragging,
      model,
      row,
      idx
    } = this.props;
    const opacity = isDragging ? 0 : 1;

    const content = (
      <div style={{
        opacity: opacity
      }}>
        <span
          itemID={'handle'}
          className={classes.dragHandle}
          onMouseDown={this.onMouseDownOnHandle}
        >
          <DragHandle color={'primary'} />
        </span>
        <div className={classes.rowContainer}>
          <div className={classNames(classes.rowItem, classes.questionText)}>
            <EditableHtml
              imageSupport={imageSupport}
              autoWidthToolbar
              disableUnderline
              label={'label'}
              markup={row.title}
              onChange={this.onRowTitleChange(idx)}
              className={classes.editor}
            />
          </div>
          {row.values.map((rowValue, rowIdx) => (
            <div key={rowIdx} className={classes.rowItem}>
              {model.responseType === 'radio' ? (
                <Radio
                  onChange={this.onRowValueChange(idx, rowIdx)}
                  checked={rowValue === true}
                />
              ) : (
                <Checkbox
                  onChange={this.onRowValueChange(idx, rowIdx)}
                  checked={rowValue === true}
                />
              )}
            </div>
          ))}
          <div className={classes.deleteIcon}>
            <Button onClick={this.onDeleteRow(idx)}>
              <Delete className={classes.deleteIcon} />
            </Button>
          </div>
        </div>
        <hr className={classes.separator} />
      </div>
    );

    return connectDragSource(connectDropTarget(content));
  }
}
const styles = theme => ({
  actions: {
    padding: 0,
    justifyContent: 'space-between'
  },
  choice: {
    padding: theme.spacing.unit,
    overflow: 'visible'
  },
  dragHandle: {
    cursor: 'move'
  },
  dragDisabled: {
    cursor: 'inherit'
  },

  container: {
    marginTop: theme.spacing.unit * 2,
    display: 'flex',
    flexDirection: 'column'
  },
  rowContainer: {
    marginTop: theme.spacing.unit * 2,
    display: 'flex',
    alignItems: 'center',
    flex: 1
  },
  rowItem: {
    flex: 1,
    display: 'flex',
    justifyContent: 'center',
    '&> div': {
      width: '100%'
    }
  },
  deleteIcon: {
    flex: 0.5,
    minWidth: '88px',
  },
  questionText: {
    flex: 2,
    display: 'flex',
    justifyContent: 'flex-start'
  },
  separator: {
    marginTop: theme.spacing.unit * 2,
    border: 0,
    borderTop: '2px solid lightgray',
    width: '100%'
  }
});

const StyledRow = withStyles(styles)(Row);

const NAME = 'row-config';

export const choiceSource = {
  canDrag() {
    return canDrag;
  },
  beginDrag(props) {
    return {
      id: props.row.id,
      index: props.idx
    };
  }
};

const StyledSource = DragSource(NAME, choiceSource, (connect, monitor) => ({
  connectDragSource: connect.dragSource(),
  isDragging: monitor.isDragging()
}))(StyledRow);

export const choiceTarget = {
  hover() {
    log('[hover]');
  },
  drop(props, monitor) {
    const item = monitor.getItem();
    log('[drop] item: ', item, 'didDrop?', monitor.didDrop());
    props.onMoveRow(item.index, props.idx);
  }
};

const StyledSourceAndTarget = DropTarget(NAME, choiceTarget, connect => ({
  connectDropTarget: connect.dropTarget()
}))(StyledSource);

export default StyledSourceAndTarget;
