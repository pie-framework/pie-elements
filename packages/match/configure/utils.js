export const generateValidationMessage = (model) => {
  const { choiceMode } = model;

  const baselineValidationMessage =
    '\nThere should be at least 1 question row.' + '\nRow and column headings should be non-blank and unique.';
  const correctResponseMessage =
    choiceMode === 'radio'
      ? '\nThere should be a correct response defined for every row.'
      : '\nThere should be at least one correct response defined for every row.';

  return 'Validation requirements:' + baselineValidationMessage + correctResponseMessage;
};
