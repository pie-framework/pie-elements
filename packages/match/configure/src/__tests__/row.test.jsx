import { shallow } from 'enzyme';
import React from 'react';
import { Row, choiceSource, choiceTarget, canDrag } from '../row';
import { defaultProps } from './configure.test';

describe('choice', () => {
  let onDeleteRow;
  let onChange;
  let connectDragSource;
  let connectDropTarget;
  let onMoveRow;

  beforeEach(() => {
    onChange = jest.fn();
    onDeleteRow = jest.fn();
    connectDragSource = jest.fn();
    connectDropTarget = jest.fn();
    onMoveRow = jest.fn();
  });

  const defaultModel = defaultProps.model;

  const wrapper = extras => {
    const props = {
      classes: {},
      model: defaultModel,
      row: {
        id: 1,
        title: 'Question Text 1',
        values: [false, false]
      },
      idx: 1,
      isDragging: false,
      onDeleteRow: onDeleteRow,
      onChange: onChange,
      connectDragSource: connectDragSource,
      connectDropTarget: connectDropTarget,
      onMoveRow: onMoveRow,
      ...extras
    };

    return shallow(<Row {...props} />);
  };

  describe('snapshot', () => {
    it('renders', () => {
      const w = wrapper();
      expect(w).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    describe('onRowTitleChange', () => {
      it('calls onChange', () => {
        const w = wrapper();
        const callback = w.instance().onRowTitleChange(1);

        callback({
          target: {
            value: 'Foo'
          }
        });

        expect(onChange).toBeCalledWith({
          rows: [
            {
              id: 1,
              title: 'Foo',
              values: [false, false]
            },
            ...defaultModel.rows.slice(1)
          ],
          ...defaultModel
        });
      });
    });

    describe('onRowValueChange', () => {
      it('calls onChange', () => {
        const w = wrapper();
        const callback = w.instance().onRowValueChange(1, 1);

        callback({
          target: {
            checked: true
          }
        });

        expect(onChange).toBeCalledWith({
          rows: [
            {
              id: 1,
              title: 'Question Text 1',
              values: [true, false]
            },
            ...defaultModel.rows.slice(1)
          ],
          ...defaultModel
        });
      });
    });

    describe('onDeleteRow', () => {
      it('calls onChange', () => {
        const w = wrapper();
        const callback = w.instance().onDeleteRow(1);

        callback();

        expect(onDeleteRow).toBeCalledWith(1);
      });
    });

    describe('onMouseDownOnHandle', () => {
      it('change canDrag value to true', () => {
        const w = wrapper();

        w.instance().onMouseDownOnHandle();

        expect(canDrag).toEqual(true);
      });
    });

    describe('onMouseUpOnHandle', () => {
      it('change canDrag value to false', () => {
        const w = wrapper();

        w.instance().onMouseUpOnHandle();

        expect(canDrag).toEqual(false);
      });
    });
  });
});

describe('spec', () => {
  let onDeleteRow;
  let onChange;
  let connectDragSource;
  let connectDropTarget;
  let onMoveRow;

  beforeEach(() => {
    onChange = jest.fn();
    onDeleteRow = jest.fn();
    connectDragSource = jest.fn();
    connectDropTarget = jest.fn();
    onMoveRow = jest.fn();
  });

  const defaultModel = {
    id: '1',
    element: 'match-element',
    rows: [{
      id: 1,
      title: 'Question Text 1',
      values: [false, false]
    }, {
      id: 2,
      title: 'Question Text 2',
      values: [false, false]
    }, {
      id: 3,
      title: 'Question Text 3',
      values: [false, false]
    }, {
      id: 4,
      title: 'Question Text 4',
      values: [false, false]
    }],
    lockChoiceOrder: false,
    partialScoring: [],
    layout: 3,
    headers: ['Column 1', 'Column 2', 'Column 3'],
    choiceMode: 'radio',
    feedback: {
      correct: {
        type: 'none',
        default: 'Correct'
      },
      partial: {
        type: 'none',
        default: 'Nearly'
      },
      incorrect: {
        type: 'none',
        default: 'Incorrect'
      }
    },
  };

  const wrapper = extras => {
    const props = {
      classes: {},
      model: defaultModel,
      row: {
        id: 1,
        title: 'Question Text 1',
        values: [false, false]
      },
      idx: 1,
      isDragging: false,
      onDeleteRow: onDeleteRow,
      onChange: onChange,
      connectDragSource: connectDragSource,
      connectDropTarget: connectDropTarget,
      onMoveRow: onMoveRow,
      ...extras
    };

    return shallow(<Row {...props} />);
  };

  describe('canDrag', () => {
    it('returns true after mousedown event was emmited on the Drag Handle', () => {
      const w = wrapper();

      w.instance().onMouseDownOnHandle();

      const result = choiceSource.canDrag({});

      expect(result).toEqual(true);
    });

    it('returns false after mouseup event was emmited on the document', () => {
      const w = wrapper();

      w.instance().onMouseUpOnHandle();

      const result = choiceSource.canDrag({});

      expect(result).toEqual(false);
    });
  });

  describe('beginDrag', () => {
    it('returns the id', () => {
      const result = choiceSource.beginDrag({ row: { id: 1 }, idx: 1 });

      expect(result).toEqual({ id: 1, index: 1 });
    });
  });

  describe('drop', () => {
    let monitor;
    let item;

    beforeEach(() => {
      item = {
        index: 1
      };
      monitor = {
        didDrop: jest.fn().mockReturnValue(false),
        getItem: jest.fn().mockReturnValue(item)
      };
    });

    it('calls onMoveRow when one row is dropped over another', () => {
      const props = {
        idx: 2,
        onMoveRow: jest.fn()
      };
      choiceTarget.drop(props, monitor);
      expect(props.onMoveRow).toBeCalledWith(1, 2);
    });
  });
});
