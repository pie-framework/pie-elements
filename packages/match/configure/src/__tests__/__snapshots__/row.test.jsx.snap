// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerConfigBlock render snapshot 1`] = `
<div
  className="AnswerConfigBlock-container-8"
>
  <WithStyles(Typography)
    component="div"
    type="body1"
  >
    Click on the labels to edit or remove. Set the correct answers by clicking each correct answer per row.
  </WithStyles(Typography)>
  <div
    className="AnswerConfigBlock-rowTable-13"
    style={Object {}}
  >
    <div
      className="AnswerConfigBlock-rowContainer-9"
    >
      <div
        className="AnswerConfigBlock-rowItem-10 AnswerConfigBlock-questionText-12"
        key="0"
      >
        <Component
          activePlugins={Array []}
          autoWidthToolbar={true}
          className="AnswerConfigBlock-headerInput-14"
          label="column label"
          languageCharactersProps={
            Array [
              Object {
                "language": "spanish",
              },
              Object {
                "language": "special",
              },
            ]
          }
          markup="Column 1"
          onChange={[Function]}
          pluginProps={
            Object {
              "audio": Object {
                "disabled": true,
              },
              "image": Object {
                "disabled": false,
              },
              "textAlign": Object {
                "disabled": true,
              },
              "video": Object {
                "disabled": true,
              },
            }
          }
        />
      </div>
      <div
        className="AnswerConfigBlock-rowItem-10"
        key="1"
      >
        <Component
          activePlugins={Array []}
          autoWidthToolbar={true}
          className="AnswerConfigBlock-headerInput-14"
          label="column label"
          languageCharactersProps={
            Array [
              Object {
                "language": "spanish",
              },
              Object {
                "language": "special",
              },
            ]
          }
          markup="Column 2"
          onChange={[Function]}
          pluginProps={
            Object {
              "audio": Object {
                "disabled": true,
              },
              "image": Object {
                "disabled": false,
              },
              "textAlign": Object {
                "disabled": true,
              },
              "video": Object {
                "disabled": true,
              },
            }
          }
        />
      </div>
      <div
        className="AnswerConfigBlock-rowItem-10"
        key="2"
      >
        <Component
          activePlugins={Array []}
          autoWidthToolbar={true}
          className="AnswerConfigBlock-headerInput-14"
          label="column label"
          languageCharactersProps={
            Array [
              Object {
                "language": "spanish",
              },
              Object {
                "language": "special",
              },
            ]
          }
          markup="Column 3"
          onChange={[Function]}
          pluginProps={
            Object {
              "audio": Object {
                "disabled": true,
              },
              "image": Object {
                "disabled": false,
              },
              "textAlign": Object {
                "disabled": true,
              },
              "video": Object {
                "disabled": true,
              },
            }
          }
        />
      </div>
      <div
        className="AnswerConfigBlock-deleteIcon-11"
      />
    </div>
    <DropTarget(DragSource(WithStyles(Row)))
      idx={0}
      inputConfiguration={
        Object {
          "audio": Object {
            "disabled": true,
          },
          "image": Object {
            "disabled": false,
          },
          "textAlign": Object {
            "disabled": true,
          },
          "video": Object {
            "disabled": true,
          },
        }
      }
      key="0"
      mathMlOptions={
        Object {
          "mmlEditing": false,
          "mmlOutput": false,
        }
      }
      maxImageHeight={150}
      maxImageWidth={300}
      minQuestions={2}
      model={
        Object {
          "choiceMode": "radio",
          "element": "match-element",
          "feedback": Object {
            "correct": Object {
              "default": "Correct",
              "type": "none",
            },
            "incorrect": Object {
              "default": "Incorrect",
              "type": "none",
            },
            "partial": Object {
              "default": "Nearly",
              "type": "none",
            },
          },
          "feedbackEnabled": true,
          "headers": Array [
            "Column 1",
            "Column 2",
            "Column 3",
          ],
          "id": "1",
          "layout": 3,
          "lockChoiceOrder": true,
          "partialScoring": Array [],
          "prompt": "Prompt",
          "rows": Array [
            Object {
              "id": 1,
              "title": "Question Text 1",
              "values": Array [
                false,
                false,
              ],
            },
            Object {
              "id": 2,
              "title": "Question Text 2",
              "values": Array [
                false,
                false,
              ],
            },
            Object {
              "id": 3,
              "title": "Question Text 3",
              "values": Array [
                false,
                false,
              ],
            },
            Object {
              "id": 4,
              "title": "Question Text 4",
              "values": Array [
                false,
                false,
              ],
            },
          ],
        }
      }
      onChange={[MockFunction]}
      onDeleteRow={[MockFunction]}
      onMoveRow={[Function]}
      row={
        Object {
          "id": 1,
          "title": "Question Text 1",
          "values": Array [
            false,
            false,
          ],
        }
      }
    />
    <DropTarget(DragSource(WithStyles(Row)))
      idx={1}
      inputConfiguration={
        Object {
          "audio": Object {
            "disabled": true,
          },
          "image": Object {
            "disabled": false,
          },
          "textAlign": Object {
            "disabled": true,
          },
          "video": Object {
            "disabled": true,
          },
        }
      }
      key="1"
      mathMlOptions={
        Object {
          "mmlEditing": false,
          "mmlOutput": false,
        }
      }
      maxImageHeight={150}
      maxImageWidth={300}
      minQuestions={2}
      model={
        Object {
          "choiceMode": "radio",
          "element": "match-element",
          "feedback": Object {
            "correct": Object {
              "default": "Correct",
              "type": "none",
            },
            "incorrect": Object {
              "default": "Incorrect",
              "type": "none",
            },
            "partial": Object {
              "default": "Nearly",
              "type": "none",
            },
          },
          "feedbackEnabled": true,
          "headers": Array [
            "Column 1",
            "Column 2",
            "Column 3",
          ],
          "id": "1",
          "layout": 3,
          "lockChoiceOrder": true,
          "partialScoring": Array [],
          "prompt": "Prompt",
          "rows": Array [
            Object {
              "id": 1,
              "title": "Question Text 1",
              "values": Array [
                false,
                false,
              ],
            },
            Object {
              "id": 2,
              "title": "Question Text 2",
              "values": Array [
                false,
                false,
              ],
            },
            Object {
              "id": 3,
              "title": "Question Text 3",
              "values": Array [
                false,
                false,
              ],
            },
            Object {
              "id": 4,
              "title": "Question Text 4",
              "values": Array [
                false,
                false,
              ],
            },
          ],
        }
      }
      onChange={[MockFunction]}
      onDeleteRow={[MockFunction]}
      onMoveRow={[Function]}
      row={
        Object {
          "id": 2,
          "title": "Question Text 2",
          "values": Array [
            false,
            false,
          ],
        }
      }
    />
    <DropTarget(DragSource(WithStyles(Row)))
      idx={2}
      inputConfiguration={
        Object {
          "audio": Object {
            "disabled": true,
          },
          "image": Object {
            "disabled": false,
          },
          "textAlign": Object {
            "disabled": true,
          },
          "video": Object {
            "disabled": true,
          },
        }
      }
      key="2"
      mathMlOptions={
        Object {
          "mmlEditing": false,
          "mmlOutput": false,
        }
      }
      maxImageHeight={150}
      maxImageWidth={300}
      minQuestions={2}
      model={
        Object {
          "choiceMode": "radio",
          "element": "match-element",
          "feedback": Object {
            "correct": Object {
              "default": "Correct",
              "type": "none",
            },
            "incorrect": Object {
              "default": "Incorrect",
              "type": "none",
            },
            "partial": Object {
              "default": "Nearly",
              "type": "none",
            },
          },
          "feedbackEnabled": true,
          "headers": Array [
            "Column 1",
            "Column 2",
            "Column 3",
          ],
          "id": "1",
          "layout": 3,
          "lockChoiceOrder": true,
          "partialScoring": Array [],
          "prompt": "Prompt",
          "rows": Array [
            Object {
              "id": 1,
              "title": "Question Text 1",
              "values": Array [
                false,
                false,
              ],
            },
            Object {
              "id": 2,
              "title": "Question Text 2",
              "values": Array [
                false,
                false,
              ],
            },
            Object {
              "id": 3,
              "title": "Question Text 3",
              "values": Array [
                false,
                false,
              ],
            },
            Object {
              "id": 4,
              "title": "Question Text 4",
              "values": Array [
                false,
                false,
              ],
            },
          ],
        }
      }
      onChange={[MockFunction]}
      onDeleteRow={[MockFunction]}
      onMoveRow={[Function]}
      row={
        Object {
          "id": 3,
          "title": "Question Text 3",
          "values": Array [
            false,
            false,
          ],
        }
      }
    />
    <DropTarget(DragSource(WithStyles(Row)))
      idx={3}
      inputConfiguration={
        Object {
          "audio": Object {
            "disabled": true,
          },
          "image": Object {
            "disabled": false,
          },
          "textAlign": Object {
            "disabled": true,
          },
          "video": Object {
            "disabled": true,
          },
        }
      }
      key="3"
      mathMlOptions={
        Object {
          "mmlEditing": false,
          "mmlOutput": false,
        }
      }
      maxImageHeight={150}
      maxImageWidth={300}
      minQuestions={2}
      model={
        Object {
          "choiceMode": "radio",
          "element": "match-element",
          "feedback": Object {
            "correct": Object {
              "default": "Correct",
              "type": "none",
            },
            "incorrect": Object {
              "default": "Incorrect",
              "type": "none",
            },
            "partial": Object {
              "default": "Nearly",
              "type": "none",
            },
          },
          "feedbackEnabled": true,
          "headers": Array [
            "Column 1",
            "Column 2",
            "Column 3",
          ],
          "id": "1",
          "layout": 3,
          "lockChoiceOrder": true,
          "partialScoring": Array [],
          "prompt": "Prompt",
          "rows": Array [
            Object {
              "id": 1,
              "title": "Question Text 1",
              "values": Array [
                false,
                false,
              ],
            },
            Object {
              "id": 2,
              "title": "Question Text 2",
              "values": Array [
                false,
                false,
              ],
            },
            Object {
              "id": 3,
              "title": "Question Text 3",
              "values": Array [
                false,
                false,
              ],
            },
            Object {
              "id": 4,
              "title": "Question Text 4",
              "values": Array [
                false,
                false,
              ],
            },
          ],
        }
      }
      onChange={[MockFunction]}
      onDeleteRow={[MockFunction]}
      onMoveRow={[Function]}
      row={
        Object {
          "id": 4,
          "title": "Question Text 4",
          "values": Array [
            false,
            false,
          ],
        }
      }
    />
    <WithStyles(AddRowButton)
      onAddClick={[MockFunction]}
    />
  </div>
</div>
`;

exports[`choice snapshot renders 1`] = `""`;
